frontend/src/components/contacts.tsx

```diff
@@ -0,0 +1,261 @@
+            autoComplete="on"
+            required
+            value={email}
+            onChange={handleEmailChange}
+          ></input>
+
+          <label htmlFor="phone_number">Phone number</label>
+          <input
+            className="phone_number"
+            name="phone_number"
+            type="text"
+            autoComplete="on"
+            required
+            value={phoneNumber}
+            onChange={handlePhoneNumberChange}
+          ></input>
+
+          <label htmlFor="notes">Extra Notes</label>
+          <input
+            className="notes"
+            name="notes"
+            type="text"
+            autoComplete="on"
+            required
+            value={notes}
+            onChange={handleNotesChange}
+          ></input>
+
+          <label>
+            Relationship with Person
+            <input
+              type="text"
+              value={relationshipType}
+              onChange={(event) => setRelationshipType(event.target.value)}
+            />
+          </label>
+          <button onClick={addRelationship}>Add</button>
+          {relationships.map((type, index) => (
+            <div key={index}>
+              <input
+                type="text"
+                value={type.type}
+                onChange={(event) =>
+                  handleRelationshipChange(index, "type", event.target.value)
+                }
+              />
+              <button onClick={() => handleRemoveRelationshipType(index)}>
+                x
+              </button>
+            </div>
+          ))}
+
+          <label>
+            Date:
+            <input
+              type="text"
+              value={eventDate}
+              onChange={(event) => setEventDate(event.target.value)}
+            />
+          </label>
+          <label>
+            Event:
+            <input
+              type="text"
+              value={eventType}
+              onChange={(event) => setEventType(event.target.value)}
+            />
+          </label>
+          <button onClick={addImportantEvent}>Add</button>
+          {importantDates.map((date, index) => (
+            <div key={index}>
+              <input
+                type="text"
+                value={date.date}
+                onChange={(event) =>
+                  handleImportantEventChange(index, "date", event.target.value)
+                }
+              />
+              <input
+                type="text"
+                value={date.event}
+                onChange={(event) =>
+                  handleImportantEventChange(index, "event", event.target.value)
+                }
+              />
+              <button onClick={() => handleRemoveImportantEvent(index)}>
+                x
+              </button>
+            </div>
+          ))}
+        </div>
+        <div>
+          <button className="add_contact_button" type="submit">
+            Add contact
+          </button>
+        </div>
+      </form>
+    </div>
+  );
+};
```

Comment:
I'm okay w/ merging this if Hyun is okay with it. My only comment is that right now we have free text inputs for event type and relationships and that we're not enforcing a date type to the input (i.e. I can make the date `foo`).

I'm okay with this for now but we may need to revisit this in the future otherwise. 

But good work on getting the dates to sync up! I know that was a little tricky :) 

tldr: form has some "invalid" values that we may want to prevent 
<img width="845" alt="image" src="https://user-images.githubusercontent.com/16482829/223025407-8a55ab34-182b-491c-a97b-f8af1c7df69f.png">
<img width="538" alt="image" src="https://user-images.githubusercontent.com/16482829/223025529-e5c9d8b6-c30d-4b52-9726-6055b14d5381.png">

Reply:
Sent some questions about this on discord.